homeassistant/components/acmeda/cover.py

```diff
@@ -0,0 +1,230 @@
+"""Support for Acmeda Roller Blinds."""
+import asyncio
+import logging
+from time import monotonic
+
+import aiopulse
+import async_timeout
+
+from homeassistant.components.cover import ATTR_POSITION, CoverDevice
+
+from .base import AcmedaBase
+from .const import DOMAIN
+from .helpers import remove_devices
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Acmeda Rollers from a config entry."""
+    hub = hass.data[DOMAIN][config_entry.data["host"]]
+    cur_covers = {}
+    # cur_groups = {}
+
+    # Acmeda updates all covers via a single API call.
```

Comment:
This seems like a good use case for `DataUpdateCoordinator`

https://developers.home-assistant.io/docs/integration_fetching_data/#coordinated-single-api-poll-for-data-for-all-entities

Reply:
Not sure I understand. The api is fully asynchronous and isn’t polled. There is a callback to notify hass that the entity has been updated.

Reply:
At first glace it looked like this was updating data from the hub.  I see now that you are sending a batch of updates.